File: Sources/Domain/Entities/Expense.swift
import Foundation

struct Expense: Identifiable, Codable {
    let id: UUID
    let budgetId: UUID
    var amount: Double
    var description: String
    var date: Date
    
    init(id: UUID = UUID(), budgetId: UUID, amount: Double, description: String, date: Date = Date()) {
        self.id = id
        self.budgetId = budgetId
        self.amount = amount
        self.description = description
        self.date = date
    }
}

File: android/app/src/main/java/com/budgetmanager/domain/entity/Expense.kt
package com.budgetmanager.domain.entity

import java.time.LocalDate
import java.time.LocalDateTime

/**
 * Expense represents a spending transaction against a budget envelope
 */
data class Expense(
    val id: Long = 0,
    val budgetId: Long,
    val amount: Double,
    val description: String,
    val date: LocalDate = LocalDate.now(),
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now()
)
